### ISSUE TITLE
OP_CHECKBLOCKATHEIGHT Errors during fresh sync

### ISSUE THREAD

**Issue by egyptianbman** [HorizenOfficial/zen#139]
### Describe the issue
During a fresh sync of the chain, I see many errors in the log output around date 2018-06-19:
```
script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification fail$d. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995$cript/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATH$IGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non$final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: S$lver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification fai$ed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 40399$script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKAT$EIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is no$-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: $olver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification fa$led. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 4039$5script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKA$HEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is n$n-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp:
Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification f$iled. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403$95script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCK$THEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is $on-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp$ Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification $ailed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 40$995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOC$ATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is
non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cp$: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification
failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 4$3995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLO$KATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction i$ non-final. nHeight: 403995script/standard.cpp: Solver: OP_CHECKBLOCKATHEIGHT verification failed. Transaction is non-final. nHeight: 403995ERROR: AcceptToMemoryPool: nonstandard transaction: scriptpubkey
```

As well as **many**:
```
ERROR: AcceptToMemoryPool: joinsplit requirements not met
```

#### If so, please list the steps to reproduce below:
1. Started wallet and saw errors
2. Deleted chain files `~/.zen/{blocks,cert.pem,chainstate,db.log,debug.log,fee_estimates.dat,key.pem,peers.dat}`
3. Start node, same errors around mid 2018.

### Expected behaviour
I expect not to see so many errors during sync

### Actual behaviour + errors
Errors as mentioned above

### The version of Zen you were using:
Zen Daemon version v2.0.15-9d4cbfa18

### Machine specs:
- OS name + version: Ubuntu 18.04
- CPU: 12 core
- RAM: 16GB
- Disk size: 2TB
- Disk Type (HD/SDD): HD
- Linux kernel version (uname -a): Linux 59ea8478bb13 4.16.12-200.fc27.x86_64 #1 SMP Fri May 25 21:10:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Compiler version (gcc -version): gcc version 7.3.0 (Ubuntu 7.3.0-16ubuntu3)

**Comment by zennermac** [comment]
Are you still seeing these errors with the latest build, Zen 2.0.17

**Comment by egyptianbman** [comment]
I haven't needed to sync a new node since I created this issue. Feel free to resolve it if you think it's resolved in 2.0.17.
